### ISSUE TITLE
werkzeug server hangs when POSTing data

### ISSUE THREAD

**Issue by philipbjorge** [timothycrosley/hug#424]
We are using werkzeug as a development server because of its great debugging capabilities.
Unfortunately, POST requests are completely non-functional and cause the server to hang until we cancel our request.

I've tracked this down to the body.read that occurs here https://github.com/timothycrosley/hug/blob/develop/hug/input_format.py#L37
Here's is werkzeug's take on the matter -- http://werkzeug.pocoo.org/docs/0.11/request_data/

Any thoughts on a workable solution to this?

**Comment by philipbjorge** [comment]
Hanging was caused by the newrelic WSGI wrapper.
There are still issues with how werkzeug handles the input stream -- https://github.com/falconry/falcon/issues/469

**Comment by philipbjorge** [comment]
tl;dr: Falcon and hugs handles everything fine and the problem lies with newrelic's wsgi wrapper and its interaction with werkzeug.
Closing the issue.
